File: LLMeetBackend/src/main/java/com/videomeeting/domain/Meeting.java
package com.videomeeting.domain;

import jakarta.persistence.*;
import lombok.Builder;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;
import java.time.LocalDateTime;

@Entity
@Table(name = "meeting")
@Data
public class Meeting {
    @Id
    @GeneratedValue(generator = "custom_id")
    @GenericGenerator(name = "custom_id", strategy = "com.videomeeting.utils.CustomeIdGenerator")
    @Column(name = "meeting_id")
    private String meetingId;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @ManyToOne
    @JoinColumn(name = "creator_id", referencedColumnName = "user_id")
    private User creator;

    @Column(name = "status")
    private String status; // "ACTIVE" or "ENDED"

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "start_time")
    private LocalDateTime startTime;

    @Column(name = "end_time")
    private LocalDateTime endedAt;

    public Meeting() {}


}

File: client/src/types/model/meeting.ts
export interface Meeting {
  meeting_id: string;
  title: string;
  description: string;
  start_time: string; // ISO 格式时间字符串
  end_time: string; // ISO 格式时间字符串
  creator_id: string;
  created_at: string; // ISO 格式时间字符串
  status: 'ready' | 'ongoing' | 'ended' | string; // 可根据后端状态值扩展
  minutes: string | null; // 会议纪要路径或内容，如果为空则为 null
}
